reference/errors/error_codes.mdx

```diff
@@ -197,6 +197,22 @@ There is an error in the provided index format, check out our guide on [index cr
 
 The [`primaryKey`](/reference/api/indexes#body-2) field is invalid. It should either be a string or set to `null`.
 
+## `invalid_multi_search_query_federated`
+
+A multi-search query includes `federationOptions` but the top-level `federation` object is `null` or missing.
+
+## `invalid_multi_search_query_pagination`
+
+A multi-search query contains `page`, `hitsPerPage`, `limit` or `offset`, but the top-level federation object is not `null`.
+
+## `invalid_multi_search_weight`
+
+A multi-search query contains a negative value for `federated.weight`.
+
+## `invalid_multi_search_queries_ranking_rules`
+
+Two or more queries in a multi-search request have incompatible results.
```

Comment:
The documentation does not explain what "incompatible results" mean, and what it means for results to be compatible.

I think it should be added, for two reasons:

1. users planning to use the feature can be aware of the need for the queries to yield compatible results
2. users encountering the error are given a better way to understand the problem and its possible resolution

[Explanation on compatible results](https://meilisearch.notion.site/v1-10-federated-search-698dfe36ab6b4668b044f735fb40f0b2#4696f7fa2685420dad4c017af7efd5ef)